src/status/libstatus/settings.nim

```diff
@@ -24,14 +24,16 @@ proc getSettings*(useCached: bool = true): JsonNode =
   dirty = false
   result = settings
 
-proc getSetting*[T](name: string, defaultValue: T, useCached: bool = true): T =
+proc getSetting*[T](name: Setting, defaultValue: T, useCached: bool = true): T =
   let settings: JsonNode = getSettings(useCached)
-  if not settings.contains(name):
+  if not settings.contains($name) or $settings{$name} == "{}":
```

Comment:
What do you think of having this function in some utils file and use it to check if a json attribute is empty or not instead of this comparison?

```nim
import json
import tables 
proc isEmpty(a: JsonNode): bool =
  case a.kind:
  of JObject: return a.fields.len == 0
  of JArray: return a.elems.len == 0
  of JString: return a.str == ""
  of JNull: return true
  else:
    return false
```

The line would then change to
```nim
if not settings.contains($name) or isEmpty(settings{$name})
```

Reply:
Thanks so much for this! Your solution is **much** better than mine 😂